src/util/debug-logger.ts

```diff
@@ -0,0 +1,73 @@
+enum LogLevel {
+  Info = 'INFO',
+  Warn = 'WARN',
+  Error = 'ERROR',
+  Debug = 'DEBUG',
+}
+let plugin: string | undefined = undefined;
+
+const originalConsole = {
+  log: console.log,
+  info: console.info,
+  warn: console.warn,
+  error: console.error,
+  debug: console.debug,
+};
+
+/**
+ * Overrides console methods with custom debug logging.
+ */
+const overrideConsoleMethods = (debugMode: boolean) => {
+  console.log = (...args: any[]) => debugLog(LogLevel.Info, args, debugMode);
+  console.info = (...args: any[]) => debugLog(LogLevel.Info, args, debugMode);
+  console.warn = (...args: any[]) => debugLog(LogLevel.Warn, args, debugMode);
+  console.error = (...args: any[]) => debugLog(LogLevel.Error, args, debugMode);
+  console.debug = (...args: any[]) => debugLog(LogLevel.Debug, args, debugMode);
+};
+
+/**
+ * Sets the name of the currently executing plugin.
+ */
+const getExecutingPluginName = (pluginName: string) => {
```

Comment:
seems it's setting the plugin name instead of getting

BTW it's better to incapsulate plugin variable, and these methods into single function